### PR TITLE
optimizer hints: add docs for hint SET_VAR

### PR DESCRIPTION
**Pull request by winoros**
### What is changed, added or deleted? (Required)

ref to https://github.com/pingcap/tidb/pull/45893

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/15013
- Other reference link(s): 

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: optimizer-hints.md [hunk 1]
```diff
@@ -83,6 +83,16 @@ This hint specifies the outer `SELECT` query block's name to `QB1`, which makes
 >
 > In the above example, if the hint specifies the `QB_NAME` to `sel_2` and does not specify a new `QB_NAME` for the original second `SELECT` query block, then `sel_2` becomes an invalid name for the second `SELECT` query block.
 
+### SET_VAR(VAR_NAME=VAR_VALUE)
+
+`SET_VAR(VAR_NAME=VAR_VALUE)` can change the value of the specified system variable. This hint can control most variables related with optimizer and executor. Refer to Chapter [System Variables](/system-variables.md) too see which variables is controlled by this hint.
```
Comment (songrijie) [inline optimizer-hints.md:88]:
```suggestion
`SET_VAR(VAR_NAME=VAR_VALUE)` can change the value of the specified system variable at statement runtime without changing the value in current session. This hint can control most variables related with optimizer and executor. Refer to Chapter [System Variables](/system-variables.md) too see which variables is controlled by this hint.
```

File: optimizer-hints.md [hunk 2]
```diff
@@ -83,6 +83,16 @@ This hint specifies the outer `SELECT` query block's name to `QB1`, which makes
...
+
+```sql
+SELECT /*+ SET_VAR(MAX_EXECUTION_TIME=1234) */ @@MAX_EXECUTION_TIME;
```
Comment (songrijie) [inline optimizer-hints.md:91]:
```suggestion
SELECT /*+ SET_VAR(MAX_EXECUTION_TIME=1234) */ @@MAX_EXECUTION_TIME;
SELECT @@MAX_EXECUTION_TIME;
```

File: optimizer-hints.md [hunk 3]
```diff
@@ -83,6 +83,16 @@ This hint specifies the outer `SELECT` query block's name to `QB1`, which makes
...
+```
+
+Executing the above SQL. The returned value is `1234` which is set by the hint, not the default value of the variable `MAX_EXECUTION_TIME`.
```
Comment (songrijie) [inline optimizer-hints.md:94]:
```suggestion
Executing the above SQL. The returned value is `1234` which is set by the hint, not the default value of the variable `MAX_EXECUTION_TIME`. The value is restored in next statement. 
```

File: system-variables.md [hunk 4]
```diff
@@ -43,12 +43,18 @@ SET GLOBAL tidb_distsql_scan_concurrency = 10;
 >
 > Variables using the same unit might compete for the same set of resources.
 
+## Hint SET_VAR
+
+TiDB introduces the Opitmizer Hint [SET_VAR](/optimizer-hints.md#SET_VAR(VAR_NAME=VAR_VALUE)) to control the execution behavior of one SQL more flexiable. This hint can change the value of specified variables at SQL level.
```
Comment (songrijie) [inline system-variables.md:48]:
```suggestion
TiDB introduces the Opitmizer Hint [SET_VAR](/optimizer-hints.md#SET_VAR(VAR_NAME=VAR_VALUE)) that changes the value of the specified system variable at statement runtime without changing the value in current session.  "Applies to Hint [SET_VAR](/optimizer-hints.md#SET_VAR(VAR_NAME=VAR_VALUE)): Yes" indicates the particular variable can be set by [SET_VAR](/optimizer-hints.md#SET_VAR(VAR_NAME=VAR_VALUE)). 
```

**Review by songrijie — COMMENTED** [review]
Rest LGTM

**Review by Oreoxmt — APPROVED** [review]

File: system-variables.md [hunk 5]
```diff
@@ -43,12 +43,25 @@ SET GLOBAL tidb_distsql_scan_concurrency = 10;
...
+- For variables with the `Applies to hint SET_VAR: Yes` setting, you can use the [`SET_VAR`](/optimizer-hints.md#set_varvar_namevar_value) hint to modify the value of the system variable in the current session during statement execution.
+- For variables with the `Applies to hint SET_VAR: No` setting, you cannot use the [`SET_VAR`](/optimizer-hints.md#set_varvar_namevar_value) hint to modify the value of the system variable in the current session during statement execution.
+
+For more information about the `SET_VAR` hint, see [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value).
+
+## Hint SET_VAR
+
+TiDB introduces the Opitmizer Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) to control the execution behavior of one SQL more flexiable. This hint can change the value of specified variables at SQL level.
+The following chapter will list which variables can be controlled by the hint.
+
```
Comment (Oreoxmt) [inline system-variables.md:57]:
```suggestion
```

File: system-variables.md [hunk 6]
```diff
@@ -5298,6 +5685,8 @@ Internally, the TiDB parser transforms the `SET TRANSACTION ISOLATION LEVEL [REA
 
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
+- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
 - Type: Boolean
 - Default value: `ON`
 - This variable controls whether to use the high precision mode when computing the window functions.
+
```
Comment (Oreoxmt) [inline system-variables.md:5692]:
```suggestion
```

File: optimizer-hints.md [hunk 7]
```diff
@@ -83,6 +83,38 @@ This hint specifies the outer `SELECT` query block's name to `QB1`, which makes
 >
 > In the above example, if the hint specifies the `QB_NAME` to `sel_2` and does not specify a new `QB_NAME` for the original second `SELECT` query block, then `sel_2` becomes an invalid name for the second `SELECT` query block.
 
+### SET_VAR(VAR_NAME=VAR_VALUE)
+
+You can use `SET_VAR(VAR_NAME=VAR_VALUE)` to temporarily modify the value of system variables during statement execution. After the statement is executed, the value of the system variable in the current session is automatically changed back to the original value. This hint can be used to modify some system variables related to the optimizer and executor. You can find the system variables that can be modified using the hint `SET_VAR()` in [System variables](/system-variables.md).
```
Comment (ran-huang) [inline optimizer-hints.md:88]:
```suggestion
You can temporarily modify the value of system variables during statement execution by using the `SET_VAR(VAR_NAME=VAR_VALUE)` hint. After the statement is executed, the value of the system variable in the current session is automatically restored to the original value. This hint can be used to modify some system variables related to the optimizer and executor. For a list of system variables that can be modified using this hint, refer to [System variables](/system-variables.md).
```

**Review by ran-huang — APPROVED** [review]

**Pull request merged**